src/llmq/quorums_signing_shares.cpp

```diff
@@ -176,23 +176,32 @@ CSigSharesInv CBatchedSigShares::ToInv() const
 
 CSigSharesManager::CSigSharesManager()
 {
-    StartWorkerThread();
+    workInterrupt.reset();
 }
 
 CSigSharesManager::~CSigSharesManager()
 {
-    StopWorkerThread();
 }
 
 void CSigSharesManager::StartWorkerThread()
 {
+    // can't start new thread if we have one running already
+    if (workThread.joinable()) {
+        assert(false);
+    }
+
     workThread = std::thread(&TraceThread<std::function<void()> >,
         "sigshares",
         std::function<void()>(std::bind(&CSigSharesManager::WorkThreadMain, this)));
 }
 
 void CSigSharesManager::StopWorkerThread()
 {
+    // make sure to call InterruptWorkerThread() first
```

Comment:
any reason we're not simply calling `InterruptWorkerThread` here?

Reply:
Technically, won't hurt calling it here too but I wanted to make sure that shutdown flow (interrupt -> stop) is correct in all cases and I didn't miss anything.